sympy/solvers/solveset.py

```diff
@@ -192,6 +191,11 @@ def inv(trig):
                 invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
             return _invert_real(f.args[0], invs, symbol)
 
+    if (isinstance(a,log) for a in f.args):
```

Comment:
> wanted to check if there is any log present or not

You can use `expr.has(log)` to do this, but you should only exclude `log` that have the symbol of interest, something like `if any(symbol in l.free_symbols for l in expr.atoms(log)):` (assuming `symbol` is the symbol being solved for and `expr` is what you are testing to see if it is log-free or not.